AssemblerLib/MeshComponentMap.cpp

```diff
@@ -34,18 +34,32 @@ MeshComponentMap::MeshComponentMap(
     : _num_components(components.size())
 
 {
+    // get number of unknows
+    GlobalIndexType num_unknowns = 0;
+    for (auto const c : components)
+    {
+        assert (c != nullptr);
+        for (unsigned mesh_subset_index = 0; mesh_subset_index < c->size(); mesh_subset_index++)
+        {
+            MeshLib::MeshSubset const& mesh_subset = c->getMeshSubset(mesh_subset_index);
+           const MeshLib::NodePartitionedMesh &mesh
+                   = static_cast<const MeshLib::NodePartitionedMesh&>(mesh_subset.getMesh());
```

Comment:
Should there be a comment that with Petsc we always work with `NodePartitionedMesh`? In a way that's natural.